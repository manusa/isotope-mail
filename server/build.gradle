buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
