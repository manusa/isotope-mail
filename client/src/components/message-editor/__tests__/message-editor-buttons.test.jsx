import React from 'react';
import {shallow} from 'enzyme/build/index';
import MessageEditorButtons from '../message-editor-buttons';


describe('MessageEditorButtons component test suite', () => {
  let commonProps;
  beforeAll(() => {
    commonProps = {
      editor: {
        selection: {
          getNode: jest.fn(() => ({}))
        }
      },
      editorState: {
        bold: true
      },
      parentSetState: jest.fn()
    };
  });
  test('Snapshot render, should render MessageEditorButtons', () => {
    // Given
    const props = {...commonProps};
    // When
    const messageEditorButtons = shallow(<MessageEditorButtons {...props} />);
    // Then
    expect(messageEditorButtons).toMatchSnapshot();
  });
  describe('Component events', () => {
    test('onToggle, should trigger button toggleFunction', () => {
      // Given
      const messageEditorButtons = shallow(<MessageEditorButtons {...commonProps} />);
      // When
      messageEditorButtons.find('MceButton[icon="link"]').props().onToggle();
      // Then
      expect(commonProps.parentSetState).toHaveBeenCalledTimes(1);
    });
  });
});
