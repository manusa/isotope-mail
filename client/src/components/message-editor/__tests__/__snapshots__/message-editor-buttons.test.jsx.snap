// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageEditorButtons component test suite Snapshot render, should render MessageEditorButtons 1`] = `
<div
  className="mdc-card message-editor__button-container"
>
  <MceButton
    active={true}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_bold"
    iconClassName="message-editor__button-icon"
    key="bold"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_italic"
    iconClassName="message-editor__button-icon"
    key="italic"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_underline"
    iconClassName="message-editor__button-icon"
    key="underline"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_list_bulleted"
    iconClassName="message-editor__button-icon"
    key="UL"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_list_numbered"
    iconClassName="message-editor__button-icon"
    key="OL"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_indent_decrease"
    iconClassName="message-editor__button-icon"
    key="outdent"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_indent_increase"
    iconClassName="message-editor__button-icon"
    key="indent"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_clear"
    iconClassName="message-editor__button-icon"
    key="clear_format"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon={null}
    iconClassName="message-editor__button-icon"
    key="H1"
    label="H1"
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon={null}
    iconClassName="message-editor__button-icon"
    key="H2"
    label="H2"
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon={null}
    iconClassName="message-editor__button-icon"
    key="H3"
    label="H3"
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="format_quote"
    iconClassName="message-editor__button-icon"
    key="blockquote"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="space_bar"
    iconClassName="message-editor__button-icon"
    key="PRE"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="code"
    iconClassName="message-editor__button-icon"
    key="code"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="link"
    iconClassName="message-editor__button-icon"
    key="link"
    label=""
    onToggle={[Function]}
  />
  <MceButton
    active={false}
    activeClassName="message-editor__button--active"
    className="message-editor__button"
    icon="link_off"
    iconClassName="message-editor__button-icon"
    key="unlink"
    label=""
    onToggle={[Function]}
  />
</div>
`;
